cli/ingress-controller/main.go

```diff
@@ -392,14 +404,40 @@ func main() {
 				Client: kongClient,
 			},
 		}
+		var cert tls.Certificate
+		if cliConfig.AdmissionWebhookCertPath != "" && cliConfig.AdmissionWebhookCert != defaultAdmissionWebhookCertPath {
+			glog.Fatalf("both --admission-webhook-cert-file and --admission-webhook-cert" +
+				"are set. Please remove one or the other")
+		}
+		if cliConfig.AdmissionWebhookKeyPath != "" && cliConfig.AdmissionWebhookKey != defaultAdmissionWebhookKeyPath {
+			glog.Fatalf("both --admission-webhook-cert-key and --admission-webhook-key" +
+				"are set. Please remove one or the other")
+		}
+		if cliConfig.AdmissionWebhookCert != "" {
+			var err error
+			cert, err = tls.X509KeyPair([]byte(cliConfig.AdmissionWebhookCert), []byte(cliConfig.AdmissionWebhookKey))
+			if err != nil {
+				glog.Fatalf("failed to load admission webhook cert: %s", err)
+			}
+		}
+		if cliConfig.AdmissionWebhookCertPath != "" && cliConfig.AdmissionWebhookCert == "" {
```

Comment:
The second condition is redundant. Due to earlier checks in the code, at this point in execution, you can be sure that  `cliConfig.AdmissionWebhookCert == ""`.

Reply:
Superfluous but helps with reading a bit--the link back to the check to confirm that only one is set isn't immediately obvious, and without the check it looks like it will overwrite the certs, even though it doesn't. Minor quibble though, so I've removed it.

Reply:
Did you actually remove it? It shows up in the current diff.

Another edge case here:
If a user mounts the cert on file-system and also sets the env vars, then the certs on the file-system take precedence here. Is this intentional? I'd recommend to have the certs in env var take precedence over path based cert because that's how I've seen precedence take place.

Reply:
It would probably behoove me to push commits after making them.

Thinking over this again in the morning re the second comment, I remembered I added it originally. This should prefer the envvar certs always, but the default path cert complicates that.

1. The envvar cert is set, and the path cert is the default. The initial sanity checks succeed.
2. The envvar cert is successfully loaded.
3. The path cert block is evaluated. With the check, this does nothing, because the envvar setting is nonempty and must have successfully loaded the cert, so we skip this block. Without the check, the path is nonempty because of the default, so we run the block and overwrite the cert.

We can place the path block first and allow the envvar block to override any cert set in the path block, but that's complicated by the potential for either block to fail fatally. I don't think it's likely for the default to fail (if it happens, you probably have bigger problems), but it can theoretically block loading of valid configuration.

Restored the check with an explanatory comment. We could alternately handle this by placing the path block first and emit a fatal error only if `admission-webhook-cert` is empty, but IMO both options are equally awkward.

Reply:
Good call. Resolving and proceeding with merge now.
We don't have any unit testing around this and based on the bugs and corner-cases we have discussed, we should have added tests to assert the right behavior.
Not a blocker, merging now.